http-client/src/main/java/io/micronaut/http/client/netty/DefaultHttpClientBuilder.java

```diff
@@ -0,0 +1,268 @@
+        ArgumentUtils.requireNonNull("nettyClientSslBuilder", nettyClientSslBuilder);
+        this.nettyClientSslBuilder = nettyClientSslBuilder;
+        return this;
+    }
+
+    @NonNull
+    DefaultHttpClientBuilder codecRegistry(@NonNull MediaTypeCodecRegistry codecRegistry) {
+        ArgumentUtils.requireNonNull("codecRegistry", codecRegistry);
+        this.codecRegistry = codecRegistry;
+        return this;
+    }
+
+    @NonNull
+    DefaultHttpClientBuilder handlerRegistry(@NonNull MessageBodyHandlerRegistry handlerRegistry) {
+        ArgumentUtils.requireNonNull("handlerRegistry", handlerRegistry);
+        this.handlerRegistry = handlerRegistry;
+        return this;
+    }
+
+    @NonNull
+    DefaultHttpClientBuilder webSocketBeanRegistry(@NonNull WebSocketBeanRegistry webSocketBeanRegistry) {
+        ArgumentUtils.requireNonNull("webSocketBeanRegistry", webSocketBeanRegistry);
+        this.webSocketBeanRegistry = webSocketBeanRegistry;
+        return this;
+    }
+
+    @NonNull
+    DefaultHttpClientBuilder requestBinderRegistry(@NonNull RequestBinderRegistry requestBinderRegistry) {
+        ArgumentUtils.requireNonNull("requestBinderRegistry", requestBinderRegistry);
+        this.requestBinderRegistry = requestBinderRegistry;
+        return this;
+    }
+
+    @NonNull
+    DefaultHttpClientBuilder eventLoopGroup(@Nullable EventLoopGroup eventLoopGroup) {
+        this.eventLoopGroup = eventLoopGroup;
+        return this;
+    }
+
+    @NonNull
+    DefaultHttpClientBuilder socketChannelFactory(@NonNull ChannelFactory<? extends Channel> socketChannelFactory) {
+        ArgumentUtils.requireNonNull("socketChannelFactory", socketChannelFactory);
+        this.socketChannelFactory = socketChannelFactory;
+        return this;
+    }
+
+    @NonNull
+    DefaultHttpClientBuilder udpChannelFactory(@NonNull ChannelFactory<? extends Channel> udpChannelFactory) {
+        ArgumentUtils.requireNonNull("udpChannelFactory", udpChannelFactory);
+        this.udpChannelFactory = udpChannelFactory;
+        return this;
+    }
+
+    @NonNull
+    DefaultHttpClientBuilder clientCustomizer(@NonNull NettyClientCustomizer clientCustomizer) {
+        ArgumentUtils.requireNonNull("clientCustomizer", clientCustomizer);
+        this.clientCustomizer = clientCustomizer;
+        return this;
+    }
+
+    @NonNull
+    DefaultHttpClientBuilder informationalServiceId(@Nullable String informationalServiceId) {
+        this.informationalServiceId = informationalServiceId;
+        return this;
+    }
+
+    @NonNull
+    DefaultHttpClientBuilder conversionService(@NonNull ConversionService conversionService) {
+        ArgumentUtils.requireNonNull("conversionService", conversionService);
+        this.conversionService = conversionService;
+        return this;
+    }
+
+    @NonNull
+    DefaultHttpClientBuilder resolverGroup(@Nullable AddressResolverGroup<?> resolverGroup) {
```

Comment:
none of these builder methods are `public`, how will it be used from the oracle cloud module?

Reply:
Only the public ones are needed. That's the point, I want to make sure the visibility modifier shows which ones we can break and which ones we cant

Reply:
I'm also confused by the non-public methods. Before you can set `conversionService` in the constructor but now it's not possible

Reply:
Yes I removed options that weren't used. If you use the constructor, now only ConversionService.SHARED is possible, except for the managed DefaultHttpClient from the context.

If there's code that needs to set the ConversionService but cant use the managed DefaultHttpClient, we can make the setter public. The idea here is that we know which properties are set by outside projects and which ones are only used with default settings.